packages/hardhat/contracts/strategy/CrabMigration.sol

```diff
@@ -254,6 +254,7 @@ contract CrabMigration is Ownable {
             IERC20(wPowerPerp).approve(crabV1, type(uint256).max);
             CrabStrategy(crabV1).withdraw(crabV1Balance);
         } else if (FLASH_SOURCE(_callSource) == FLASH_SOURCE.FLASH_MIGRATE_V1_TO_V2) {
+            uint256 initialCrabAmount = CrabStrategyV2(crabV2).balanceOf(address(this));
```

Comment:
The best way to do it might be to avoid using balanceOf comepletely in all calculations and just do proper accounting internally if we can pass in amount to transfer. This is a decent fix for now

Reply:
I feel like the way we have changed it to now is safer in a way vs reimplementing the logic of CrabStrategyV2 to have internal accounting.